api/src/main/java/org/openmrs/module/openconceptlab/client/OclClient.java

```diff
@@ -42,11 +42,15 @@
 import java.util.zip.ZipInputStream;
 
 public class OclClient {
-	
+
 	public static final String DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss'Z'";
-	
+
 	public static final String FILE_NAME_FORMAT = "yyyyMMdd_HHmmss";
-	
+
+	public static final String OCL_API_BASE_URL = "api.openconceptlab.org";

```

Comment:
What happens when this is changed to something like `api.qa.openconceptlab.org`? https://github.com/openmrs/openmrs-ocl-client/search?q=api.openconceptlab.org&unscoped_q=api.openconceptlab.org

Do we have to recompile the module?

Reply:
Thinking about it, I see two options of working this out
1. Adding the server base url as a global variable
That means having a field above the subscription url where the use specifies the server url and default one being  this, **api.openconceptlab.org**
2. Extracting this base url from the subscription url by extracting the sub url before /users 
Which one would you advice?

Reply:
@jecihjoy which do you think is better?

Reply:
Another option to get the url part after base url is
URL url = new URL(subscriptionUrl);
subUrlAfterBaseUrl = url.getPath;

In this case we wouldn't care to know anything about the server base url.

I was thinking this is the best option, what is your option?

Reply:
I do not like a solution which complicates life for the end users. 😊

Reply:
However I have already saved the server base url as a global variable, if it makes sense I could retain the code but use the above getPath method.

Reply:
What i do not like is users having to break the url into two parts whenever they come to this screen. That is what i call a complicated user interface.

Reply:
So go with any solution where the user interface stays as it was, and change of the api server does not require us to recompile the module.

Reply:
That makes sense, actually I was not having the user provide the two url part as in the the screenshot I shared, I had gone the path of getting the host url form the subscription url provided and saving it as a global variable